docs/developer-guide/transform-a-stream-with-ksql.rst

```diff
@@ -0,0 +1,92 @@
+.. _transform-a-stream-with-ksql:
+
+Transform a Stream With KSQL 
+############################
+
+KSQL enables *streaming transformations*, which you can use to convert
+streaming data from one format to another in real time. With a streaming
+transformation, not only is all existing data in a stream converted, but so
+is every record that subsequently arrives on the source stream.
+
+Transform a Stream's Properties

```

Comment:
I think the intention is to describe that the WITH clause (in the example below) can be used for a variety of things, right?

If so, I'm not sure whether "a Stream's Properties" is very descriptive.  I mean, I get it, but effectively there are [five things](https://docs.confluent.io/current/ksql/docs/developer-guide/syntax-reference.html#create-stream-as-select) you can do:

- Change the data format (today: for message values; in the future: for both keys and values).
- Change the number of partitions.
- Change the number of replicas.
- Change the timestamp field and/or the timestamp format.
- Change the new stream's underlying Kafka topic name.

I am wondering whether we should call these out explicitly, rather than hiding them all under the rather non-descriptive name of "Properties"?  I think most readers will not really know what "the properties of a stream" are.